Executor skeleton

borditamas opened the pull request.

borditamas committed:
init executor skeleton

borditamas committed:
executor skeleton impl

borditamas commented on pom.xml:
```diff
@@ -28,12 +28,14 @@
 	
-	<repositories>
+	
+	<repositories> <!-- TODO: delete this after new app library release  -->
```
remember to delete!

borditamas commented on pom.xml:
```diff
@@ -42,6 +44,11 @@
 			<artifactId>arrowhead-application-library-java-spring</artifactId>
 			<version>4.4.0.0</version>
+		</dependency>
+		<dependency> <!-- TODO: delete this after new app library release  -->
```
remember to delete!

borditamas committed:
executor skeleton impl

borditamas committed:
executor skeleton cert

rbocsi commented on application-skeleton-executor/src/main/java/eu/arrowhead/application/skeleton/executor/controller/ExecutorController.java:
```diff
@@ -0,0 +1,58 @@
+import eu.arrowhead.common.dto.shared.ChoreographerExecutorServiceInfoResponseDTO;
+
+@RestController
+public class ExecutorController {
```
a TODO here about the baseUri, like
`// @RequestMapping("/executor")  // TODO: specify the base URI here`

rbocsi commented on application-skeleton-executor/src/main/java/eu/arrowhead/application/skeleton/executor/controller/ExecutorController.java:
```diff
@@ -0,0 +1,58 @@
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	@PostMapping(path = CommonConstants.CHOREOGRAPHER_EXECUTOR_CLIENT_SERVICE_START_URI, produces = MediaType.APPLICATION_JSON_VALUE)
```
consumes instead of produces

rbocsi commented on application-skeleton-executor/src/main/java/eu/arrowhead/application/skeleton/executor/controller/ExecutorController.java:
```diff
@@ -0,0 +1,58 @@
+	
+	//-------------------------------------------------------------------------------------------------
+	@PostMapping(path = CommonConstants.CHOREOGRAPHER_EXECUTOR_CLIENT_SERVICE_ABORT_URI, produces = MediaType.APPLICATION_JSON_VALUE)
+	public void abort(@RequestBody final ChoreographerAbortStepRequestDTO request) {
```
consumes instead of produces

rbocsi commented on application-skeleton-executor/src/main/java/eu/arrowhead/application/skeleton/executor/controller/ExecutorController.java:
```diff
@@ -0,0 +1,58 @@
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	//TODO: implement here your provider related REST end points
```
provider???

rbocsi reviewed (changes requested):
see inline comments

borditamas committed:
executor skeleton controller updates

borditamas committed:
executor skeleton execution support impl

borditamas committed:
executor skeleton execution add missing property

rbocsi commented on application-skeleton-executor/src/main/java/eu/arrowhead/application/skeleton/executor/execution/ExecutionBoard.java:
```diff
@@ -0,0 +1,97 @@
+		
+		synchronized (lock) {
+			final Job job = new Job(jobRequest, ExecutionSignal.DO);
+			board.put(getUinqueIdentifier(jobRequest), job);
```
typo: Uinque => Unique

rbocsi commented on application-skeleton-executor/src/main/java/eu/arrowhead/application/skeleton/executor/execution/ExecutionBoard.java:
```diff
@@ -0,0 +1,97 @@
+	
+	//-------------------------------------------------------------------------------------------------
+	public Optional<Job> peekJob(final long sessionId, final long sessionStepId) {
+		final Job job = board.get(getUinqueIdentifier(sessionId, sessionStepId));
```
same here

rbocsi commented on application-skeleton-executor/src/main/java/eu/arrowhead/application/skeleton/executor/execution/worker/UnkownServiceExecutionWorker.java:
```diff
@@ -0,0 +1,33 @@
+					 job.getJobRequest().getSessionId(),
+					 job.getJobRequest().getSessionStepId(),
+					 job.getJobRequest().getMainOrchestrationResult().getService().getServiceDefinition());
+	}
```
should notify the choreographer about the error

rbocsi reviewed (changes requested):
see inline comments

borditamas committed:
executor skeleton fix

borditamas committed:
check configuration

borditamas committed:
Some fix and Merge branch 'development' into executor-skeleton

# Conflicts:
#	application-skeleton-provider/src/main/java/eu/arrowhead/application/skeleton/provider/ProviderApplicationInitListener.java
#	application-skeleton-publisher/src/main/java/eu/arrowhead/application/skeleton/publisher/PublisherApplicationInitListener.java
#	application-skeleton-subscriber/src/main/java/eu/arrowhead/application/skeleton/subscriber/SubscriberApplicationInitListener.java
#	pom.xml

borditamas commented:
**Status update:** Waiting for Core v4.5.0 release (App-lib will contain Choreographer realated DTOs)
